tests/utilities/test_enums.py

```diff
@@ -0,0 +1,11 @@
+from pytorch_lightning.utilities import DeviceType
+
+
+def test_consistency():
+    assert DeviceType.TPU not in ('GPU', 'CPU')
+    assert DeviceType.TPU in ('TPU', 'CPU')
+    assert DeviceType.TPU in ('tpu', 'CPU')
+    assert DeviceType.TPU not in {'GPU', 'CPU'}
+    # hash cannot be case invariant
+    assert DeviceType.TPU not in {'TPU', 'CPU'}
```

Comment:
this is so unintuitive as it works for tuple `assert DeviceType.TPU in ('TPU', 'CPU')`

Reply:
I mean, what can you do about it. It is a limitation of Python.

`Tuple` checks the `__eq__` of the values, whereas `Set` checks the `__eq__` of the `__hash__` of the values.